storage/encrypt.go

```diff
@@ -0,0 +1,236 @@
+// Copyright © 2018 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package storage
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/clearlinux/clr-installer/cmd"
+	"github.com/clearlinux/clr-installer/errors"
+	"github.com/clearlinux/clr-installer/log"
+)
+
+const (
+	// MinPassphraseLength is the shortest possible password
+	MinPassphraseLength = 8
+	// MaxPassphraseLength is the shortest possible password
+	MaxPassphraseLength = 94
+
+	// RequiredBundle the bundle needed if encrypted partitions are used
+	//RequiredBundle  = "boot-encrypted"
+
+	// RequiredBundle1 needs to be replaced with RequiredBundle
+	RequiredBundle1 = "bootloader-extras"
+	// RequiredBundle2 needs to be replaced with RequiredBundle
+	RequiredBundle2 = "storage-utils"
+
+	// EncryptHash use for LUKS encryption
+	EncryptHash = "sha256"
+	// EncryptCipher use for LUKS encryption
+	EncryptCipher = "aes-xts-plain64"
+	// EncryptKeySize use for LUKS encryption
+	EncryptKeySize = 512
+)
+
+// EncryptionRequiresPassphrase checks all partition to see if encryption was enabled
+func (bd *BlockDevice) EncryptionRequiresPassphrase() bool {
+	enabled := (bd.Type == BlockDeviceTypeCrypt && bd.FsType != "swap")
+
+	for _, ch := range bd.Children {
+		if len(ch.Children) > 0 {
+			enabled = enabled || ch.EncryptionRequiresPassphrase()
+		} else {
+			enabled = enabled || (ch.Type == BlockDeviceTypeCrypt &&
+				ch.FsType != "swap")
```

Comment:
can we replace all these ```ch.FsType != "swap"``` for a function call ?

Reply:
there are only three occurrences, one each in three different source files. You think making that logic into a function call makes it more readable?

Reply:
makes it more expandable

Reply:
@dorileo Updated in latest push